### ISSUE TITLE
Apps reattach after reboot. Logs empty.

### ISSUE THREAD

**Issue by Double-Crunch** [j-hc/zygisk-detach#55]
**Are you sure you read the [readme](https://github.com/j-hc/zygisk-detach#usage)?**
Yes

**Android version and skin (MIUI etc... is MIUI still a thing? idk):**
![Screenshot_20241117-042222](https://github.com/user-attachments/assets/41104910-86a0-4304-b8e1-4c419a5b39cf)
![Screenshot_20241117-042159](https://github.com/user-attachments/assets/9aec4cc8-85ce-4057-9ab0-31c2d4fe1f69)

**Screenshot of Play Store page of the app you are trying to detach:**
Doesn't matter

**Screenshot of the helper CLI showing detached apps (from Option 2):**
![Screenshot_20241117-042206](https://github.com/user-attachments/assets/c4602acc-0938-4b4f-bb76-3d2546272755)

**Logs taken with `su -c logcat -d -s "zygisk-detach" > /sdcard/log.txt`:**
Logs always come out empty.

Not providing this information will get your issue closed

After reboot, all apps get re-attached and I have to detach them again. Log file comes out empty.

**Comment by CRCinAU** [comment]
I've noticed this too - it seems like sometimes even without a reboot, things with just 'fix' themselves.

If I go back into the app, the previously detached apps will now not show as detached...

**Comment by j-hc** [comment]
if logs are empty, this has nothing to do with zygisk-detach because it wasnt even run.
the problem is with your setup. i cant do anything about that.

**Comment by CRCinAU** [comment]
> if logs are empty, this has nothing to do with zygisk-detach because it wasnt even run. the problem is with your setup. i cant do anything about that.

Is there anything we can do besides just installing the module in Magisk?

That's all I've ever done in the past.

**Comment by j-hc** [comment]
> > if logs are empty, this has nothing to do with zygisk-detach because it wasnt even run. the problem is with your setup. i cant do anything about that.
> 
> Is there anything we can do besides just installing the module in Magisk?

dont add play store in the denylist

**Comment by CRCinAU** [comment]
I don't have Play Store in any deny lists... What I've seemed to figure out, if I load the detach-app, I select YT, YT Music & Photos and hit Detach... Then close all apps, open up Play Store, check for updates, and YT + YT Music will show an update available.

If I then close all apps again, re-open the detach-app, select the same apps and hit detach again, then open Play Store, hit Check for Updates, and the detached apps no longer show an update.

It almost seems that when Play Store loads, it resets the detach process.

In the Magisk blocklist for Shamiko, I have:
* Google Play Services
* Google Wallet

The "Enforce DenyList" in Magisk is disabled so Shamiko handles hiding root...

**Comment by j-hc** [comment]
> I don't have Play Store in any deny lists... What I've seemed to figure out, if I load the detach-app, I select YT, YT Music & Photos and hit Detach... Then close all apps, open up Play Store, check for updates, and YT + YT Music will show an update available.
> 
> If I then close all apps again, re-open the detach-app, select the same apps and hit detach again, then open Play Store, hit Check for Updates, and the detached apps no longer show an update.
> 
> It almost seems that when Play Store loads, it resets the detach process.
> 
> In the Magisk blocklist for Shamiko, I have:
> 
>     * Google Play Services
> 
>     * Google Wallet
> 
> 
> The "Enforce DenyList" in Magisk is disabled so Shamiko handles hiding root...
> 
> Again, there is nothing in the logs when I do any of this.

nothing in the logs > module is not even being loaded by magisk > the problem is with your setup > i cant do anything about it

**Comment by CRCinAU** [comment]
From looking at the magisk logs (not via logcat), and filtering for detach, see:

```
11-17 19:42:13.871   839   839 I : Magisk 28.0(28000) daemon started
11-17 19:42:13.873   839   839 I : * Device API level: 35
11-17 19:42:13.882   839   841 I : ** post-fs-data mode running
11-17 19:42:13.907   839   841 I : * Initializing Magisk environment
11-17 19:42:13.919   839   841 I : * Running post-fs-data.d scripts
11-17 19:42:13.920   839   841 I : * Setup internal mounts
11-17 19:42:13.921   839   841 I : * Running module post-fs-data scripts
11-17 19:42:13.921   846   846 I : zygisk-detach: exec [post-fs-data.sh]
11-17 19:42:13.927   846   846 I : playintegrityfix: exec [post-fs-data.sh]
11-17 19:42:13.939   839   850 I : denylist: initializing internal data structures
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.unstable]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.feedback]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.learning]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.persistent]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.remapping1]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.room]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms.ui]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms:car]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms:identitycredentials]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.gms/com.google.android.gms:snet]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.apps.walletnfcrel/com.google.android.apps.walletnfcrel]
11-17 19:42:13.939   839   850 I : denylist add: [com.google.android.apps.walletnfcrel/com.google.android.apps.walletnfcrel:primes_lifeboat]
11-17 19:42:13.940   839   850 I : denylist add: [au.com.fuel7eleven/au.com.fuel7eleven]
11-17 19:42:13.940   839   850 I : denylist add: [com.revolut.revolut/com.revolut.revolut]
11-17 19:42:13.940   839   850 I : denylist add: [com.revolut.revolut/com.revolut.revolut:onfido_process]
11-17 19:42:13.940   839   850 I : denylist add: [com.revolut.revolut/com.revolut.revolut:p84253884fe9d4ac4dee0a2b1]
11-17 19:42:13.940   839   850 I : denylist add: [com.revolut.revolut/com.revolut.revolut:revolut_login_sync]
11-17 19:42:13.940   839   850 I : denylist add: [com.roborock.smart/com.roborock.smart]
11-17 19:42:14.048   846   846 I : zygisk_shamiko: exec [post-fs-data.sh]
11-17 19:42:14.074   839   841 I : * Loading modules
11-17 19:42:14.074   839   841 I : zygisk-detach: loading mount files
11-17 19:42:15.538   839   850 I : ** late_start service mode running
11-17 19:42:15.539   839   850 I : * Running service.d scripts
11-17 19:42:15.564   839   850 I : service.d: exec [com.google.android.youtube.sh]
11-17 19:42:15.566   839   850 I : service.d: exec [com.google.android.apps.youtube.music.sh]
11-17 19:42:15.580   839   850 I : service.d: exec [com.google.android.apps.photos.sh]
11-17 19:42:15.593   839   850 I : * Running module service scripts
11-17 19:42:15.593   839   850 I : zygisk-detach: exec [service.sh]
11-17 19:42:15.614   839   850 I : playintegrityfix: exec [service.sh]
11-17 19:42:15.616   839   850 I : zygisk_shamiko: exec [service.sh]
11-17 19:42:23.508   839  1709 I : ** boot-complete triggered
```

**Comment by j-hc** [comment]
nothing from zygisk-detach

try with only zygisk-detach enabled and see

**Comment by CRCinAU** [comment]
Maybe I'm misunderstanding how things work, but aren't these lines from the detach-app?

```
11-17 19:42:15.564   839   850 I : �service.d: exec [com.google.android.youtube.sh]
11-17 19:42:15.566   839   850 I : �service.d: exec [com.google.android.apps.youtube.music.sh]
11-17 19:42:15.580   839   850 I : �service.d: exec [com.google.android.apps.photos.sh]
```

**Comment by pooyashahidi** [comment]
I have the same issue on A15 using apatch. The same setup was working on A14 btw.
I noticed the whole detatch.bin file is getting deleted after a reboot

**Comment by j-hc** [comment]
> I have the same issue on A15 using apatch. The same setup was working on A14 btw. I noticed the whole detatch.bin file is getting deleted after a reboot

do you get any logs? if yes, then its apatch removing the detach.bin while placing the modules.img

**Comment by suryadip2008** [comment]
> I have the same issue on A15 using apatch. The same setup was working on A14 btw. I noticed the whole detatch.bin file is getting deleted after a reboot

Same issue. When I was using the app to detach the apps, they were reattching themselves after reboot but when I used WebUI to detach the apps, the issue seems to have gone...

**Comment by pooyashahidi** [comment]
sorry for not having logs. there was nothing in them. but I think i figured out the issue.
since @suryadip2008 mentioned apatch next. I too use apatch next and i think that is the issue. it keeps applying the modules_update folder to the modules folder and thus deleting the bin file for this to work. deleting the modules_update folder solved this issue.

here is the test I did. I manually created some extra files there with .bin and without extension with different names to see what is happening. 
before reboot:
![Image](https://github.com/user-attachments/assets/e9aced4a-2456-4fa7-a19e-236372cfa0f8)

after reboot:
![Image](https://github.com/user-attachments/assets/b711ee49-12cc-4152-a834-1908cdc5821f)
